### PR TITLE
Feature/komarch 27 scout

### PR DESCRIPTION
**Pull request by yanchith**
This sets up scout and schema for posts for elastic.

Noteworthy:

- The wordpress import script now uses the `Searchable` trait to also update the elastic index
- Use `php artisan elastic:migrate` to create or migrate elastic indexes
- Use `php artisan scout:import "<model>"` to batch import existing records of `<model>`

This is still draft, because:

- [x] While the `php artisan scout:import` command says it completed successfully, asynchronous data ingestion in elastic actually fails and no records are imported. It fails with "stacktrace": ["org.elasticsearch.index.mapper.MapperParsingException: failed to parse field [title] of type [text] in document with id '728'. Preview of field's value: '{sk=N?mestie SNP v Martine - v?sledky s??a?e}'". I'll look into it.

This can still be reviewed meanwhile.

### CODE REVIEW

**Comment by yanchith** [comment]
Found out why elastic was crashing. Apparently we need to tell scout what fields are JSON, treating them as text makes parsing on elastic side unhappy. I think Elastic *could* parse the field, but there is a mismatch between what scout tells elastic and about the field and what the field contains.

**Comment by yanchith** [comment]
Also, tbh I don't know why CI fails. It says it cannot find Faker, but this branch is already downstream from the fix where Faker was moved to normal dependencies.

File: config/elastic.client.php [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?php declare(strict_types=1);
+
+return [
+    'hosts' => [
+        env('ELASTIC_HOST', 'localhost:9200'),
```
Comment (igor-kamil) [inline config/elastic.client.php:5]: please update also `.env.example` -> check e.g. https://github.com/SlovakNationalGallery/exlibris.pamatniknarodnihopisemnictvi.cz/blob/master/.env.example

File: README.md [hunk 2]
```diff
@@ -29,6 +29,8 @@ installs as well as long as you configure them equivalently (see
 - Run `composer install`
 - Run `npm install`
 - Run `php artisan migrate`
+- Run `php artisan elastic:migrate`
+- Run `php artisan scout:import`
```
Comment (igor-kamil) [inline README.md:33]:
```suggestion
- Run `php artisan scout:import <model>`
```

or 
```suggestion
- Run `php artisan scout:import "App\Models\Post"`
```

File: app/Http/Controllers/PostsController.php [hunk 3]
```diff
@@ -9,7 +9,15 @@ class PostsController extends Controller
...
     public function index(Request $request)
     {
-        $posts = Post::published()->orderBy('published_at', 'desc')->paginate(10);
+        $search = $request->input('search');
+
+        $posts = null;
+        if (empty($search)) {
+            $posts = Post::published()->orderBy('published_at', 'desc')->paginate(10);
+        } else {
+            $posts = Post::search($search)->orderBy('published_at', 'desc')->paginate(10);
```
Comment (igor-kamil) [inline app/Http/Controllers/PostsController.php:18]:
```suggestion
            $posts = Post::search($search)->paginate(10);
```
just a nit :D 
but keep it sorted by relevance

File: elastic/migrations/2020_12_24_003632_create_posts_index.php [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+final class CreatePostsIndex implements MigrationInterface
+{
+    /**
+     * Run the migration.
+     */
+    public function up(): void
+    {
+        Index::create('posts', function (Mapping $mapping, Settings $settings) {
+            $mapping->object('title');
```
Comment (igor-kamil) [inline elastic/migrations/2020_12_24_003632_create_posts_index.php:17]: happy to see you solved that! just for the record, in #regarch we did it in this way: https://github.com/SlovakNationalGallery/register-architektury/blob/master/app/Models/Building.php#L125-L165
This seems to work nicely so far... keep it.

**Review by igor-kamil — APPROVED** [review]
works nicely. would be nice to update also `.env.example` - but it's not blocking from the approve ;)

**Comment by yanchith** [comment]
Thanks for the package-lock regeneration! I merged it manually and must have made an error.

**Comment by yanchith** [comment]
@igor-kamil I think I addressed everything. Couldn't get the CI running, we can ask our colleagues. Going to merge, so we can close the sprint.

**Pull request merged**